src/toast/todmap/conviqt.py

```diff
@@ -507,47 +507,39 @@ def exec(self, data):
                 beam_file = self._beam_file[det]
             except TypeError:
                 beam_file = self._beam_file.replace("DETECTOR", det)
-                beam_file_i00 = beam_file.replace(".fits", "_I000.fits")
-                beam_file_0i0 = beam_file.replace(".fits", "_0I00.fits")
-                beam_file_00i = beam_file.replace(".fits", "_00I0.fits")
+                beam_file_i00= beam_file.replace('.fits', '_I000.fits')
+                beam_file_0i0= beam_file.replace('.fits', '_0I00.fits')
+                beam_file_00i= beam_file.replace('.fits', '_00I0.fits')
 
-            beamI00 = self.get_beam(beam_file_i00, det, verbose)
-            beam0I0 = self.get_beam(beam_file_0i0, det, verbose)
-            beam00I = self.get_beam(beam_file_00i, det, verbose)
+
+            beamI00 =self.get_beam(beam_file_i00, det,verbose )
+            beam0I0 =self.get_beam(beam_file_0i0, det,verbose )
+            beam00I =self.get_beam(beam_file_00i, det,verbose )
             detector = self.get_detector(det)
 
             theta, phi, psi = self.get_pointing(data, det, verbose)
+            psipol = self._get_psipol(data, det)
+
+            hwpang = self._get_hwpangle(data)
+            if hwpang is None:
+                psitot =  psipol
+            else:
+                psitot =  psipol + 2 * hwpang
+
+            # I-beam convolution
+            pnt = self.get_buffer(theta, phi, psi  , det, verbose)
```

Comment:
Instead of `get_buffer(theta, phi, psi, ..)` try `get_buffer(theta, phi, psi - psipol, ...)`

Reply:
sorry, not `psipol` but `psitot` to include the HWP